File: autojs/src/main/java/com/stardust/autojs/Pref.java
package com.stardust.autojs;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

/**
 * Created by Stardust on 2017/12/8.
 */

public class Pref {


    private static Pref sInstance;
    private SharedPreferences mSharedPreferences;

    public Pref(Context context) {
        mSharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
    }

    public static void setInstance(Pref instance) {
        if (sInstance != null)
            throw new IllegalStateException();
        sInstance = instance;
    }

    public static Pref getInstance() {
        return sInstance;
    }


}

File: autojs/src/main/java/com/stardust/autojs/core/pref/Pref.kt
package com.stardust.autojs.core.pref

import android.preference.PreferenceManager
import com.stardust.app.GlobalAppContext

object Pref {
    private val preferences = PreferenceManager.getDefaultSharedPreferences(GlobalAppContext.get())

    val isStableModeEnabled: Boolean
        get() {
            return preferences.getBoolean("key_stable_mode", false)
        }

    val isGestureObservingEnabled: Boolean
        get() {
            return preferences.getBoolean("key_gesture_observing", false)
        }
}